### PR TITLE
Replace Travis with GitHub Workflow

### PR DESCRIPTION
**Pull request by vtomole**

### CODE REVIEW

**Comment by vtomole** [comment]
@ncrubin  Please merge this. You will have to chnage the Openfermion checks in the setting for this repo to these builds instead of Travis before you do.

File: check/mypy [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# Usage:
+#     check/mypy [--flags]
+################################################################################
+
+# Get the working directory to the repo root.
+cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd "$(git rev-parse --show-toplevel)"
+
+echo -e -n "\033[31m"
+mypy --config-file=dev_tools/conf/mypy.ini $@ .
```
Comment (viathor) [inline check/mypy:15]: nit: it's safer with quotes: `"$@"` (which is equivalent to quoting each array element separately)

File: dev_tools/conf/mypy.ini [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+[mypy-__main__]
+follow_imports = silent
+ignore_missing_imports = true
+
+# 3rd-party libs for which we don't have stubs
+[mypy-apiclient.*,numpy.*,pytest.*,scipy.*,,setuptools.*,_pytest.*]
+follow_imports = silent
+ignore_missing_imports = true
+
+#Adding "sympy.* or mypy-sympy to the above list (3rd-party libs for which we don't have stubs) doesn't ignore "cannot find module 'sympy' error
```
Comment (viathor) [inline dev_tools/conf/mypy.ini:12]: nit: space after # for consistency with comment above

**Comment by vtomole** [comment]
@kevinsung  Windows tests run now.

File: check/mypy (hunk 1 above)
Comment (vtomole) [reply check/mypy]: Done

File: dev_tools/conf/mypy.ini (hunk 2 above)
Comment (vtomole) [reply dev_tools/conf/mypy.ini]: Done

**Comment by kevinsung** [comment]
Actually I'm just going to go ahead since we shouldn't leave this hanging.

**Pull request merged**